compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp

```diff
@@ -0,0 +1,386 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/RemoveUnnecessaryTransposeNetPass.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+
+#include <vector>
+
+namespace
+{
+
+class TaggedShapeAnalyzer final
+{
+public:
+  /**
+   * @brief check 'Transpose-Reshape-Transpose' can be replaced by one 'Reshape'.
+   *
+   * NOTE:
+   *  - It tracks How each dimension of shape is transformed/moved by using 'tags'.
+   *    Tag records the initial order of each dimension.
+   *
+   *  - Tag is only attached for dimensions with non-1 value.
+   *
+   *  - For details, please refer analyze_(transpose, reshape) comments.
+   */
```

Comment:
> I would be great if you can add some comments about the logic how the pass works.
https://github.com/Samsung/ONE/pull/11891#issuecomment-1797814451 

I appended an overall idea here. 
Also, I added a comment with an example for `analyze_transpose` and `analyze_reshape` functions.

But I feel it's still hard to deliver the overall logic intuitively. :thinking:  
I'm trying to find a better way to explain. Hmm..

Reply:
Add some examples